File: CPP/Algoritmo1/ExercicioLoop/exercicioLoop13.cpp
#include <iostream>
using namespace std;
int main() {
  int insert, fatorialAux, fatorial = 0, impar, imparRes = 0, fibonacci, fib1, fib2, quant = 0;
  float percentImpar;
  do {
    cout << "Informe um numero \n";
	cin >> insert;
	if (insert>0){
		fatorialAux = insert;
		fatorial = insert;
		while (fatorialAux>1){
			fatorialAux = fatorialAux - 1;
			fatorial = fatorial*fatorialAux;
		}
		cout << "Fatorial: " << fatorial;
		fibonacci = 0;
		fib1 = 0;
		fib2 = 1;
		while (fibonacci<insert){
			fibonacci = fib1 + fib2;
			fib1 = fib2;
			fib2 = fibonacci; 
    	}
		if (insert == fibonacci){
			cout << "\nPertence ao fibonacci\n";
		}else{
			cout << "\nNao pertence ao fibonacci\n";
		}
		quant++;
		impar = insert % 2;
		if (impar == 1){
			imparRes++;
		}
		cout << "\n";
	}
  } while (insert != 0);
  	percentImpar = imparRes*100/quant;
	cout << "\nQuantidade: " << quant;
	cout << "\nImpar: "<< percentImpar << "%";
}

File: Python/Fundamentos/exercicioLoop13.py
contador = 0
valor = 1
contadorImpar = 0
while valor != 0:
    valor = int(input("Insira o valor "))
    fatorialAux = valor
    fatorial = valor
    while fatorialAux > 1:
        fatorialAux = fatorialAux - 1
        fatorial = fatorial * fatorialAux
    print("Fatorial %i" % (fatorial))
    fibonacci = 0
    fib1 = 0
    fib2 = 1
    while fibonacci < valor:
        fibonacci = fib1 + fib2
        fib1 = fib2
        fib2 = fibonacci
    if valor == fibonacci:
        print("Pertence fibonacci")
    else:
        print("Não pertence fibonacci")
    contador += 1
    impar = valor % 2
    if impar == 1:
        contadorImpar += 1
percentualImpar = contadorImpar*100/contador
print("Quantidade: %i" % (contador))
print("Porcentagem de impares: %i" % (percentualImpar))
